website/static/js/logFeed.js

```diff
@@ -1,238 +1,245 @@
+                });
+                for (i = ctrl.totalPages - MAX_PAGES_ON_PAGINATOR_SIDE; i < ctrl.totalPages - 1; i++) {
+                    ctrl.paginators().push({
+                        text: i + 1,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+            }
+            // two ellipses
+            else {
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+                for (i = parseInt(ctrl.currentPage()) - 1; i <= parseInt(ctrl.currentPage()) + 1; i++) {
+                    ctrl.paginators().push({
+                        text: i + 1,
+                        url: function() {
+                            ctrl.pageToGet(parseInt(this.text));
+                            if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                                return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                            }
+                        }
+                    });/* jshint ignore:line */
+                    // function defined inside loop
+                }
+                ctrl.paginators().push({
+                    text: '...',
+                    url: function() { }
+                });
+            }
+            // last page
+            ctrl.paginators().push({
+                text: ctrl.totalPages,
+                url: function() {
+                    ctrl.pageToGet(ctrl.totalPages);
+                    if (ctrl.pageToGet() !== ctrl.currentPage()) {
+                        return _buildLogUrl(ctrl.node, ctrl.pageToGet());
+                    }
+                }
+            });
+            // next page
+            ctrl.paginators().push({
+                url: function() { return ctrl.nextPage; },
+                text: '>'
+            });
+        }
 
-////////////////
-// Public API //
-////////////////
+        return m('.db-activity-list.m-t-md', [
 
-var defaults = {
-    /** Selector for the progress bar. */
-    progBar: '#logProgressBar'
-};
+            ctrl.failed ? m('p', 'Unable to retrieve logs at this time.') :
 
+            ctrl.logRequestPending() ?  m('.spinner-loading-wrapper', [
+                m('.logo-spin.logo-lg'),
+                m('p.m-t-sm.fg-load-message', 'Loading logs...')
+            ]) :
 
-var initViewModel = function(self, logs, url){
-    self.logs = createLogs(logs);
-    self.viewModel = new LogsViewModel(self.logs, url);
-    if(url) {
-        self.viewModel.fetchResults();
+            [ctrl.activityLogs() ? ctrl.activityLogs().map(function(item) {
```

Comment:
Erm, scratch that; I see now there are multiple children nested under `.db-activity-list`, so the outer array may be necessary.